extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeParsingHelper.java

```
@@ -393,14 +393,32 @@ public static JsonArray getJsonResponse(String url, Localization localization) t
      * @throws ContentNotAvailableException if an alert is detected
      */
     public static void defaultAlertsCheck(JsonObject initialData) throws ContentNotAvailableException {
-        final JsonArray alerts = initialData.getArray("alerts");
-        if (alerts != null && !alerts.isEmpty()) {
-            final JsonObject alertRenderer = alerts.getObject(0).getObject("alertRenderer");
-            final String alertText = alertRenderer.getObject("text").getString("simpleText");
-            final String alertType = alertRenderer.getString("type");
-            if (alertType.equalsIgnoreCase("ERROR")) {
-                throw new ContentNotAvailableException("Got error: \"" + alertText + "\"");
+        /* Should ignore this alert text
+        Get high-quality audio, gapless playback and personalised music recommendations.
+        */
+        /*
+        Possibilities:
+        No alerts,
+        alerts with no errors,
+        errors with no text,
+        errors with text
+         */
+
+        try {
+            final JsonObject alertRenderer = initialData.getArray("alerts").getObject(0).getObject("alertRenderer"); // Exits here if No Alerts
+            if (alertRenderer.getString("type").equalsIgnoreCase("ERROR")) { // Often exits here With Youtube-Music // Exits here if alert has no errors
+                try {
+                    throw new ContentNotAvailableException("Got alert error: \"" + alertRenderer.getObject("text").getString("simpleText") + "\""); // Finishes here if errors have text
+                } catch (Exception ignored) {
+                    throw new ContentNotAvailableException("Got unknown alert error"); // Finishes here if errors have no text
+                }
+            } else {
+                // This condition will only be reached if we encounter an alert that has a type, but is not an error
             }
+
+        } catch (Exception ignored) {
```

This will ignore all exceptions including a ContentNotAvailableException thrown above

---

What should happen? Should we throw an error?

---

Give an example of when that type of error would be thrown.